### DISCUSSION TITLE
Help for land-sea masking

### DISCUSSION THREAD

**Discussion by SajiMohandas** [dtcenter/METplus#1145] [Configuration] [answered]
I have created a bitmapped nectdf with land-sea masking over India.

What is the command syntax to use this 'mask file' in grid_stat command
line or in Config file?

Thanks and regards
Saji

**Comment by JohnHalleyGotway** [comment, marked as answer]
Hello Saji, I see you have a question about how to use the output of the Gen-Vx-Mask tool in the MET statistics tools, like Grid-Stat. Good news, this is very easy. Just list the path to that NetCDF data file in the "mask.poly" entry of the Grid-Stat configuration file. By way of example, please take a look at this line of a sample Grid-Stat config file:
https://github.com/dtcenter/MET/blob/fb666bc3289e2bf49ccfc85e716ab812e10a7628/met/scripts/config/GridStatConfig_APCP_12#L80

The "mask.poly" entry is an array of strings which indicate the paths to files. Those can be paths to the NetCDF output of Gen-Vx-Mask or polyline files themselves. In general, we recommend running Gen-Vx-Mask first rather than reprocessing the polyline file each time you run Grid-Stat. If the polyline contains many points or the grid is dense, re-defining that mask in each run of Grid-Stat can be slow.

Here's the documentation about the mask dictionary:
https://met.readthedocs.io/en/develop/Users_Guide/config_options.html#mask

Note that Grid-Stat will read and apply the land/sea mask from that Gen-Vx-Mask output file. In the resulting .stat output files, you'll see that the VX_MASK column is set to the NetCDF variable name from the Gen-Vx-Mask output. The default variable name "data_mask" isn't very descriptive. When you run Gen-Vx-Mask, I'd recommend using the "-name" command line argument to set that output variable name to something more descriptive. And then that'll ultimately show up in the VX_MASK column of the Grid-Stat output.

**Comment by SajiMohandas** [comment]
Hi

Thanks for the clarification. But I am doing data masking and dont have a
poly file with lat/long values to create the ocean mask.

I created a mask file with the ocean values 0 and land values 1 using a
python program and used this mask file for grid_stat tool within the
config file in the format given below (Met version 7).

mask = {
   grid = [ "FULL" ];
   poly = [ "/home/saji/bigbang/LIGHTNING_verify/stats/oceanmask.nc" ];
};

It is running fine and creating grid_stat output with double output, first
with original nomask stats and below it is repeated with different values.
Is this second is the masked statistics (only over land and ocean masked).

How can we avoid writing double output stats? Or any other changes in
config file is required?

Thanks in advance
Saji

**Reply by JohnHalleyGotway** [reply]
Saji,

The Grid-Stat tool creates output for each masking region defined in the mask dictionary in the config file. By convention, when grid is set to "FULL" Grid-Stat computes statistics using all points in the verification domain. If you do not want output for the FULL verification domain, just remove it from the list:
```
mask = {
   grid = [ ];
   poly = [ "/home/saji/bigbang/LIGHTNING_verify/stats/oceanmask.nc" ];
};
```
Please refer to the documentation of the [mask dictionary](https://met.readthedocs.io/en/develop/Users_Guide/config_options.html#mask). Note that "mask" appears in the config file for multiple tools and all options are discussed in the documentation. But for Grid-Stat, only the "grid" and "poly" options apply.

**Comment by SajiMohandas** [comment]
Dear John

Thanks a lot for your confirmation. Yes this is working now.

Thanks again

Saji

**Reply by JohnHalleyGotway** [reply]
Saji, thanks for writing back to confirm. Since your question is answered, I'm going to "Lock" this discussion to prevent future updates. If more questions arise in your use of the METplus tools, please just start a new discussion topic.
